cni/network/invoker_azure.go

```diff
@@ -131,35 +138,39 @@ func (invoker *AzureIPAMInvoker) Delete(address *net.IPNet, nwCfg *cni.NetworkCo
 		nwCfg.IPAM.Subnet = invoker.nwInfo.Subnets[0].Prefix.String()
 	}
 
-	if address == nil {
+	if len(addresses) == 0 {
 		if err := invoker.plugin.DelegateDel(nwCfg.IPAM.Type, nwCfg); err != nil {
 			return invoker.plugin.Errorf("Attempted to release address with error:  %v", err)
 		}
-	} else if len(address.IP.To4()) == 4 {
-		nwCfg.IPAM.Address = address.IP.String()
-		log.Printf("Releasing ipv4 address :%s pool: %s",
-			nwCfg.IPAM.Address, nwCfg.IPAM.Subnet)
-		if err := invoker.plugin.DelegateDel(nwCfg.IPAM.Type, nwCfg); err != nil {
-			log.Printf("Failed to release ipv4 address: %v", err)
-			return invoker.plugin.Errorf("Failed to release ipv4 address: %v", err)
-		}
-	} else if len(address.IP.To16()) == 16 {
-		nwCfgIpv6 := *nwCfg
-		nwCfgIpv6.IPAM.Environment = common.OptEnvironmentIPv6NodeIpam
-		nwCfgIpv6.IPAM.Type = ipamV6
-		nwCfgIpv6.IPAM.Address = address.IP.String()
-		if len(invoker.nwInfo.Subnets) > 1 {
-			nwCfgIpv6.IPAM.Subnet = invoker.nwInfo.Subnets[1].Prefix.String()
-		}
+	}
 
-		log.Printf("Releasing ipv6 address :%s pool: %s",
-			nwCfgIpv6.IPAM.Address, nwCfgIpv6.IPAM.Subnet)
-		if err := invoker.plugin.DelegateDel(nwCfgIpv6.IPAM.Type, &nwCfgIpv6); err != nil {
-			log.Printf("Failed to release ipv6 address: %v", err)
-			return invoker.plugin.Errorf("Failed to release ipv6 address: %v", err)
+	for _, address := range addresses {
+		if len(address.IP.To4()) == bytesSize4 {
```

Comment:
Isn't 
ip.To4() != nil sufficient? to know if it's ipv6

Reply:
This logic was already in place and I think the reason they differentiate both IPv4 and IPv6 is to have the else at the end in case you pass an invalid IP. I believe there is a UT currently in place that is also meant to test Invalid IPs and I don't know if `ip.To4() != nil` is true or not in the case of invalid IP requests.